main.rb

```diff
@@ -0,0 +1,166 @@
+# Add your requires for classes here
+require_relative 'app'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+# Initialize data structures
+people = []
+books = []
+rentals = []
+
+# Define the display_menu method
+def display_menu
+  puts 'School Library App'
+  puts 'Choose an option:'
+  puts '1. List all books'
+  puts '2. List all people'
+  puts '3. Create a person'
+  puts '4. Create a book'
+  puts '5. Create a rental'
+  puts '6. List rentals for a person'
+  puts '7. Quit'
+end
+
+# Define the choice_option method
+def choice_option(option, books, people, rentals)
+  case option
+  when 1
+    list_all_books(books)
+  when 2
+    list_all_people(people)
+  when 3
+    create_person_menu(people)
+  when 4
+    create_book_menu(books)
+  when 5
+    create_rental_menu(people, books, rentals)
+  when 6
+    list_rentals_menu(rentals, people)
+  else
+    puts 'Invalid option. Please choose a valid option.'
+  end
+end
+
+# Define the main method
+def main(books, people, rentals)
+  puts 'Welcome to School Library App'
+
+  loop do
+    display_menu
+    option = gets.chomp.to_i
+    if option == 7
+      puts 'Exiting the app. Goodbye!'
+      break
+    else
+      choice_option(option, books, people, rentals)
+    end
+  end
+end
+
+def create_rental_menu(people, books, rentals)
+  loop do
+    puts 'Create a Rental'
+    person_id = choose_person(people)
+    break if person_id == 'back'
+
+    book_title = choose_book(books)
+    break if book_title == 'back'
+
+    date = enter_rental_date
+
+    create_rental(people, books, rentals, person_id.to_i, book_title, date)
```

Comment:
- [ ] When calling create_rental in this line of code you're passing 6 parameters, however the create_rental method in the app.rb file only takes 1 parameter therefore when a user tries to create a rental they receive an error, please update your code to fix this bug.